### PR TITLE
client, daemon: rework rest changes to export "ready" and "err"

### PR DESCRIPTION
**Pull request by chipaca**

### CODE REVIEW

File: client/change.go [hunk 1]
```diff
@@ -44,20 +45,8 @@ type Task struct {
...
-	var ar [2]int
-	if err := json.Unmarshal(buf, &ar); err != nil {
-		return err
-	}
-
-	tp.Done = ar[0]
-	tp.Total = ar[1]
-
-	return nil
+	Current int
```
Comment (niemeyer) [inline client/change.go]: Missing field tags (and tests?).

Comment (chipaca) [inline client/change.go]: Hmm!

Comment (chipaca) [inline client/change.go]: the tests do cover this, though. Has the `json` package gotten smarter?

File: client/change_test.go [hunk 2]
```diff
@@ -41,7 +42,37 @@ func (cs *clientSuite) TestClientChange(c *check.C) {
 		Kind:    "foo",
 		Summary: "...",
 		Status:  "Do",
-		Tasks:   []*client.Task{{Kind: "bar", Summary: "...", Status: "Do", Progress: client.TaskProgress{Done: 0, Total: 1}}},
+		Tasks:   []*client.Task{{Kind: "bar", Summary: "...", Status: "Do", Progress: client.TaskProgress{Current: 0, Total: 1}}},
```
Comment (niemeyer) [inline client/change_test.go]: Huh.. how come this is passing if the field tag is not there?

**Comment by mvo5** [comment]
:+1:

**Comment by chipaca** [comment]
https://play.golang.org/p/S9CXfy4iX0
the `json` package has gotten smarter since I tried that last (or we all misremember).
In any case, adding those field tags.

**Comment by niemeyer** [comment]
LGTM

**Pull request merged**